### ISSUE TITLE
Phenotype Profile Search - Feedback

### ISSUE THREAD

**Issue by iimpulse** [monarch-initiative/monarch-ui#258]
Analyze Phenotypes is now live in beta. Please test this vigorously and report bugs here!!

@monicacecilia @jmcmurry @kshefchek @pnrobinson 

https://beta.monarchinitiative.org/analyze/phenotypes

**Comment by pnrobinson** [comment]
It is not really obvious to me how this page works, actually. 

What do I do once I get here?

![image](https://user-images.githubusercontent.com/4943420/72446161-03115d80-3781-11ea-97c3-3872c4f3ee52.png)

**Comment by iimpulse** [comment]
Any suggestions?? @pnrobinson

**Comment by pnrobinson** [comment]
I mean, I have entered two HPO terms but there is nothing that I can do with them?

**Comment by kshefchek** [comment]
"This Phenotype Profile Search enables you search"

Should be 

"This Phenotype Profile Search enables you to search"

When creating a second profile for comparison, clicking the X to remove a phenotype removes it from the first profile (suspect something is being reused between the two).  Can walk through the steps if it's hard to replicate.

**Comment by pnrobinson** [comment]
I am completely lost -- this page really needs more documentation!

**Comment by monicacecilia** [comment]
> this page really needs more documentation!

Thanks, @pnrobinson. Agreed! We will continue working on improvements to documentation here. https://github.com/monarch-initiative/monarch-ui/issues/181

**Comment by nlharris** [comment]
Looks pretty cool! Here are some minor suggestions:

enables you search our database -> enables you to search the Monarch database

Create A Profile of Phenotypes -> Create a profile of phenotypes (or, if you insist on Title Case, it would be Create a Profile of Phenotypes -- "A" is not capitalized)

I typed "marfan" in the search box and it found 0 matches. I thought, hmm, maybe it's case-sensitive? But when I went back, it found 41 matches for marfan, and I was not able to recreate the error. Weird.

In the "Matching Strings", it would be nice if you could hide the markup:
<em class="hilite">Marfan</em> and <em class="hilite">Marfan</em>-related disorder

The matches seem like a dead end--I was expecting the matching terms to hyperlink to HPO terms, but nothing links to anything. I don't see anything else that one can do with the list of matches other than copy-paste it somewhere in its idiosyncratic format.

If you do something on the matches page (e.g., click on one of the categories) and then go back, you get back to the Phenotype Profile Search page with no search string, rather than back to the complete list of matches. (Yes, user error, but not an unlikely one.)

If I choose "Enter a phenotype list", the help text says "Enter a comma separated list of prefixed phenotype ids e.g. HP:00000322". It doesn't say they have to be HP IDs--it only says that in the error message. Also, the example doesn't show a list separated by commas, so it's not really an example of what it's asking for.

If I enter a list of HP terms without commas, the error message says "Please enter phenotype term id's from the Human Phenotype Ontology (e.g. HP:0000002)". It doesn't note that the problem here is the lack of commas, and it doesn't give an example that includes a comma.

When I got to the similarity page, it was pretty cool!
I would again hope that the info in the table would link to something, though.

**Comment by wdduncan** [comment]
I did a search for "breast cancer". The phenotype list that was returned was not very informative. How are the listed phenotypes associated with breast cancer? Do they indicate higher likelihood? If so, then some extra information would be good. For example, prostate cancer was listed as a phenotype. But, breast cancer is quite rare in men. I think it is important to communicate this.

I then compared the phenotype list to HER2 gene (NCBIGene:100054739). In the resulting matrix, it would good if good if there tooltips explaining the meaning of the column headers.

**Comment by kshefchek** [comment]
One issue I see here is that we're searching all of Mondo and getting inferred phenotype associations.  In the case of searching on Breast Cancer we're returning phenotypes from several subtypes which include duplicate terms.  We need to limit autocomplete terms on leaf nodes in Mondo, which isn't currently possible with biolink but can be added.  We could also filter anything out that doesn’t contain an OMIM equivalence.  See semi related https://github.com/monarch-initiative/mondo/issues/685

**Comment by kshefchek** [comment]
also wanted to clarify that the prostate cancer annotation is not a bug, it is annotated to Hereditary Breast And Ovarian Cancer Syndrome ORPHA:145, https://hpo.jax.org/app/browse/disease/ORPHA:145, which is a subtype of breast cancer, MONDO:0007254

**Comment by damiansm** [comment]
This before I try it:

Saw some comments above about needing more documentation and improving it which is a bad sign for a UI i.e. it should be simple enough without having to read the docs but good to have more explanation in there

Now I will try it:

- Maybe too many words in the first sentence at top - don't need to know it is powered by biolink. That can be a logo at bottom of page
- Maybe a combined pasting a list as well as being able to add to it by searching is simpler
- If you search with a phenotype term e.g. "Craniosynostosis" and it matches phenotypes and diseases only show the phenotype matches or put thenm at the top
- Results phenogrid is truncated on my Mac with firefox
- Results table does not have links out
- I still feel lost when I get a MONDO disease ID rather than OMIM or Orphanet ID. Is it possible to show both. Guess when you link out you can see which of your favourite OMIM entries it equates to

**Comment by LCCarmody** [comment]
- It is not obvious how to get back to the beginning once you have chosen (either Search & build or I have a phenotype list).  

- Also, why doesn't the field where you add HP IDS in the 'create a profile of phenotypes' just say 'Add HP IDS ONLY).  I added HP labels at first thinking that would also be acceptable.

- I also did a test case using 'classic familial adenomatous polyposis'.  There are a lot of odd phenotypes pulled up including Long philtrum, long face, hypertelorism, broad nasal tip, thick upper lip vermilion, and hypotonia.  I am pretty sure a number of those aren't correct.  I'll try to look for sources and look for other test cases, but it seems odd.

**Comment by monicacecilia** [comment]
- Explain the table of results for the Phenogrid
  - ```Source data for each match result (each column) in the Phenogrid above is represented as a row in the table below.```
- Paint the score purple.
- Remove Taxon ID column
- Filter to remove or any of the columns
- For later, download the source data file for this table @deepak, @kent, that includes the fields on the current version of the table, and to also include ‘most

**Comment by jmcmurry** [comment]
Today at leadership call we discussed this as a workflow; there was agreement that this is an easier-to-follow workflow, however I would like @pnrobinson to look.

![image](https://user-images.githubusercontent.com/3658863/73967938-b9beb480-48cd-11ea-8cf7-9f180ed1cf33.png)

**Comment by pnrobinson** [comment]
Sorry I could not make it this AM. 
I do not understand how you can start with "show me everything"-"all matching genes" -- matching to what? I  guess we mean after the page  "1. Create A Profile of Phenotypes " ?  If so, this is OK. But I would recommend that we simplify the Create Profile page. There are too many unnecessary clicks. For instance, there is no use in asking the user to confirm the profile. If they don't like it they can just go back.

**Comment by jmcmurry** [comment]
Hi Peter, thanks for jumping in. 

I know it seems unnecessary, but if you don't confirm the profile, there's no way to know that you're done and launch the analysis. It is too computationally intensive and confusing to dynamically generate a new phenogrid with ever edit to the profile. Let's schedule a live walkthrough with just you and me next week, thanks. I'll explain more then.

**Comment by pnrobinson** [comment]
Sure, let's chat next week. However, the page as is really is quite complicated and even though I have seen it multiple times it does not seem natural. I doubt that new users would like this and it seems that if there is a button saying "If you are done click here to see the phenogrid" would save one intermediate page and some confusion!

**Comment by monicacecilia** [comment]
@jmcmurry @pnrobinson - Did you reach a verdict?

**Comment by monicacecilia** [comment]
@pnrobinson's proposal is, in general terms, to simplify the tool with the recommendations listed below:
1. Restricting the first dialogue to use only 'phenotypes' to initiate the search.
   - Allow user to type a list of HPO terms
   - Allow user to choose from auto-complete options
2. Remove the option to 'hide' the list of phenotypes
3. Add an option for user to add phenotypes from a 'disease' of interest
4. Remove the buttons to confirm the list of phenotypes 
   - ([click to see](https://github.com/monarch-initiative/monarch-ui/issues/258#issuecomment-583652081) @jmcmurry's explanation above for why this may not be possible)
5. Run the analysis by comparing this list of phenotypes 'against everything' in Monarch.
6. Either better clarify what is on the table of results under the Phenogrid or remove the table all together

- Peter also shared that he considers that the current form of Phenotype Profile search prevents the release of monarch-ui v.1.0. The tool should be simplified before opening to the public.

**Comment by cmungall** [comment]
I don't have massively strong opinions. Peter's comments seem reasonable

**Comment by mellybelly** [comment]
So for right now, there is no way to otherwise compare a disease or gene entity's phenotype profile against other known entities because Phenogrid is not implemented on the disease pages. Therefore at least for now, I would not recommend we remove it- it has always felt to be one of our most used comparisons. 

Second, i think most people would prefer to start with a disease phenotype profile and add/subtract from it rather than adding it to a list of pre-entered phenotypes. 

Perhaps some improved layout/labels would be the best bet here for now? How about we flip the two functionalities left/right - e.g. "Enter a set of phenotypes" and "Compare a set of disease or gene phenotypes".  

Ideally, but perhaps not for this release, there would be the following ways to enter a set of phenotypes:
You should be able to generate phenotype terms in one of three ways: 
1 - type or paste a lump of text and do NER, confirm list.
2 - auto complete from adding names of phenotypes, confirm list.
3 - paste a list of IDs, confirm list.
4 - 'Search and Build' by grabbing phenotypes associated with a disease or gene, and then edit/confirm list.

Another request: add a tooltip for the show/hide phenotypes, i don't know what the eye icon means.

**Comment by pnrobinson** [comment]
@mellybelly  I agree these are nice features, but the current user experience is much too complicated. It would be better if there were different entries to the phenogrid on the existing page -- I like your suggestion of having a button on the disease pages. 
If we want to release the new website version promptly, then I am wondering if it would be better to restrict the features (for now!). As I understand it, this is the last block. 
Feel free to disagree, but I really think the current page is confusing!

**Comment by mellybelly** [comment]
Also - just to be realistic here- none of us are generic users, we just need to get the release out and then test it on people other than us ;-)

Also- we could have simple search and advanced search- many sites do that successfully. I personally wouldn't find it that useful to not have all the options, but i agree we don't want to scare folks off.

**Comment by mellybelly** [comment]
@monicacecilia i will enlist a few users to review the ideas here. I think it would be dangerous to remove functionality to such a large extent without doing more user testing. Remember that the monarch web app has a very diverse user base. also tagging @jmcmurry for her UX interview perspective. 

Regardless, lets try the changes i recommended and see if we can agree that it is good enough and we can iterate in the next release with improvements.

**Comment by monicacecilia** [comment]
@mellybelly also added: 

#### For Step 2: 
- To prevent possible confusion with the current labels, change the options to: 
   - 'Enter another phenotype profile'
   - 'diseases'
   - 'genes'
   - do not need to keep the ability to compare against everything
      - get thoughts about this from @pnrobinson and @jmcmurry 
- Do this in parallel construction to what was suggested for the first step. 

#### For the results table: 
- Keep the results table.
- Agreed with updating the legend for the table to say: "```Source data for each match result (each column) in the Phenogrid above is represented as a row in the table below.```"

#### For v1.1 
   - add the ability to download the table that is generated.
   - allow interplay between the grid and the table.

**Comment by pnrobinson** [comment]
So it seems we all agree that a stepwise approach is best, right? Can we try to agree on the necessary step to get the first release out the door?
As to the table -- can somebody say what it adds to the phenogrid (which is hyperlinked in case people want more information)

**Comment by cmungall** [comment]
I would rather get something working if a little complicated out sooner, and iterate on improvements later. We need to kill the old site

**Comment by monicacecilia** [comment]
After further discussion on 2020-03-02, we have all agreed to the following (and I believe the technical term is "_Quitsies, no anti-quitsies, no startsies!_"): 😉 

Dear @iimpulse please: 

- [ ] Update workflow with the decision tree shown below. 
  - Of note - request to add more examples: _patient profile to disease_ matching, _disease to gene_ matching, and _gene to disease_ matching (nice-to-have).
- [ ] Add a Reset button at the top
- [ ] If possible, do some forensics on silent and unpredictable errors reportedly experienced during testing.
- [ ] Please remember to add the [items in the list of ToDos above](https://github.com/monarch-initiative/monarch-ui/issues/258#issuecomment-582617146) (I commented out the ones that no longer apply).

Please ping @jmcmurry or I to address any questions on the monarch-ui Gitter chat.

Thank you!!!
------------
Updated decision tree:
![PPS-decisiontree-2020-03-02](https://user-images.githubusercontent.com/1714621/75762322-5454bc80-5cef-11ea-8368-cdaca1b0f0c3.png)

**Comment by pnrobinson** [comment]
No objections to moving forward like this, but honestly, can we do some user testing after this? I am probably wrong, but for me it is quite confusing and complicated and I think we run a high risk of losing more users than impressing other users with the flexibility of the tool. There really should also be just a generic "show me stuff" but on page one that will have sensible default values.

**Comment by pnrobinson** [comment]
The entry page says "Do you have a phenotype list you would like to paste?", but if you click "YES" you can either paste or type. I would suggest "Do you have a list of phenotypic features?" or "Do you have a list of phenotype terms"?

**Comment by monicacecilia** [comment]
Most requested work has been completed, leftover requests are now part of a this new ticket (#327) for Milestone v1.1.
